pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -260,8 +260,14 @@ func (exp *FlowExporter) Run(stopCh <-chan struct{}) {
 func (exp *FlowExporter) sendFlowRecords() (time.Duration, error) {
 	currTime := time.Now()
 	var expireTime1, expireTime2 time.Duration
-	exp.expiredConns, expireTime1 = exp.conntrackConnStore.GetExpiredConns(exp.expiredConns, currTime, maxConnsToExport)
+	// In scenarios involving inter-node service traffic and an egress network policy with a drop action, records are
```

Comment:
Rephrased for a bit, will this be better?

We always export records from denyConnStore first, then conntrackConnStore.
It is because only records from denyConnStore contain deny network policy
metadata. For a inter-node connection with egress drop network policy, both
conntrackConnStore and denyConnStore will send out records to Flow Aggregator.
If the record from conntrackConnStore arrives FA first, FA will not be able to
capture the deny network policy metadata, and it will keep waiting for a record
from destination node to finish flow correlation until timeout.
Later on we probably should consider doing a record deduplication between
conntrackConnStore and denyConnStore before exporting records.

Reply:
> We always export records from denyConnStore first, then conntrackConnStore.
It is because only records from denyConnStore contain deny network policy
metadata.

I think the order is only important for the `inter-node connection with egress drop network policy` case?

How about this:

For an inter-node connection with egress drop network policy, our Flow Exporter from the source Node will export records from both conntrackConnStore and denyConnStore to Flow Aggregator. If the record from conntrackConnStore arrives FA first, FA will not be able to capture the deny network policy metadata, and it will keep waiting for a record from the destination Node to finish flow correlation until timeout. It is because the correlation requirement for records from the denyConnStore and conntrackConnStore in this case are different and we only do the aggregation process if records are from the same Node.

Later on we probably should consider doing a record deduplication between
conntrackConnStore and denyConnStore before exporting records.

Reply:
We export records from denyConnStore first, then conntrackConnStore.
We enforce the ordering to handle a special case: for a inter-node connection
with egress drop network policy, both conntrackConnStore and denyConnStore
will send out records to Flow Aggregator. If the record from conntrackConnStore
arrives FA first, FA will not be able to capture the deny network policy metadata,
and it will keep waiting for a record from destination node to finish flow correlation
until timeout. Later on we probably should consider doing a record deduplication
between conntrackConnStore and denyConnStore before exporting records.

Reply:
> I think the order is only important for the inter-node connection with egress drop network policy case?

I know. I meant to handle the inter-node egress drop case, here we enforce the export ordering to always export deny conns first, then conntrack.

Reply:
The latest comments looks better to me. I've updated it. Thanks!